main/dpdk.c

```diff
@@ -66,25 +66,36 @@ static ssize_t log_write(void *, const char *buf, size_t size) {
 	return n;
 }
 
-int dpdk_init(struct gr_args *args) {
+int dpdk_log_init(const struct gr_args *args) {
 	cookie_io_functions_t log_functions = {.write = log_write};
-	char main_lcore[32] = {0};
-	char **eal_args = NULL;
-	int ret;
 
 	if (getenv("INVOCATION_ID")) {
 		// executed by systemd
 		log_syslog = true;
 		openlog("grout", LOG_PID | LOG_ODELAY, LOG_DAEMON);
 	}
 
-	if ((log_stream = fopencookie(NULL, "w+", log_functions)) == NULL) {
-		ret = errno;
-		goto end;
-	}
+	gr_rte_log_type = rte_log_register_type_and_pick_level("grout", RTE_LOG_NOTICE);
+	if (gr_rte_log_type < 0)
+		return errno_log(-gr_rte_log_type, "rte_log_register_type_and_pick_level");
+
+	if ((log_stream = fopencookie(NULL, "w+", log_functions)) == NULL)
+		return errno_log(errno, "fopencookie");
+
 	rte_openlog_stream(log_stream);
+	if (args->log_level >= RTE_LOG_DEBUG)
+		rte_log_set_global_level(RTE_LOG_DEBUG);
```

Comment:
Setting global log <level> it not equivalent to passing --log-level=*:<level>.
Doing this change will raise the global filter to <level>, but no DPDK component will have their log level touched.

IOW, if a DPDK component default log level is WARNING, then we will never get its debug level logs, while it was the case before this patch.

Reply:
I mad misunderstood what this `--log-level` option is meant for. I will change this to `rte_log_set_level_pattern("*", ...)`. Is there any reason to change the global level?

Reply:
This global level filters all logs to a certain level, regardless of what a logtype is configured with. Its use is quite limited (I see it as useless) and it is best to use rte_log_set_("*") like before.